### PR TITLE
[DON'T MERGE] support geos 3.6.1

### PR DESCRIPTION
**Pull request by filnet**
For future reference.

Compiles but has not been tested yet...

Apparenlty macos brew now offers geos 3.6.1

### CODE REVIEW

**Comment by filnet** [comment]
Note that the PR is against branch 2.7

**Comment by willysr** [comment]
Tested with osgearth 2.8 and it compiles.
+1 of merging it.

**Comment by filnet** [comment]
msys2 now also offers geso 3.6.1

**Comment by outruner** [comment]
Hello after applying this patch on 2.8 I could successfuly compile with  geos 3.6.1.
However  I have a crash on osgearth_viewer with following earth : feature_country_boundaries.earth.
it seems that this GeometryFactory is null for some reason. Can anyone reproduce this issue?

**Comment by outruner** [comment]
Hello after applying a little modification to the patch I could succcessfuly run the sample. Please read my comment on the patch for more informations (unique_ptr should be passed by reference and not by value). regards.

**Comment by filnet** [comment]
I pushed a fix for the crash.

File: src/osgEarthSymbology/GEOS.cpp [hunk 1]
```diff
@@ -66,7 +66,7 @@ namespace
     }
 
     geom::Geometry*
-    import( const Symbology::Geometry* input, const geom::GeometryFactory* f )
+    import( const Symbology::Geometry* input, const geom::GeometryFactory::unique_ptr f )
```
Comment (filnet) [inline src/osgEarthSymbology/GEOS.cpp]: i reverted to passing a pointer to the factory instead of a unique_ptr

**Comment by outruner** [comment]
thanks !

**Comment by filnet** [comment]
Tried the fix with master.

osgearth_viewer.exe src/osgearth-osgearth-2.8/tests/feature_country_boundaries.earth works with the proposed fix, but there is a segfault when exiting (on msys2 at least).

I also see a bunch of message like this :

error: linking with uncompiled shadererror: linking with uncompiled shadererror: linking with uncompiled shadererror: linking with uncompiled shader
VERTEX glCompileShader "oe_mp_vertModel" FAILED
VERTEX Shader "oe_mp_vertModel" infolog:
0:1(10): error: the compatibility profile is not supported
0:1(10): error: GLSL 3.30 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.00 ES, and 3.00 ES

**Comment by outruner** [comment]
in release mode there is no crash (fork of 2.8), compiler msvc2015. However in debug mode I also have a crash on exit (mutex.lock issue) , stacktrace attached. I did not see any shader or opengl error in both modes.
crash_osg

**Comment by filnet** [comment]
The GLSL problem is probably related to msys2.

I get the segfault with a release build but i am on msys2 not mscv.
I can't say if the mutex related segfault is related to geos (?).

**Comment by filnet** [comment]
osgearth_viewer with simple_model.earth also exhibits the segfault on exit.

So, I guess, it is safe to say that the segfault is not due to geos upgrade.

**Comment by manisandro** [comment]
Patch based on which preserves GEOS < 3.6.0 compatibility: https://github.com/manisandro/osgearth/commit/9cb5029b86a7a51ce367ffe6a190f6a26f7f418d

**Comment by filnet** [comment]
@manisandro best would be to turn your patch into a PR on master

**Comment by manisandro** [comment]
Done, I suppose this one should be closed.

**Comment by filnet** [comment]
superseded by https://github.com/gwaldron/osgearth/pull/899

**Pull request closed without merging**